src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,117 @@
+package duke;
+
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Todo;
+
+
+/**
+ * Parses in the user input and executes the corresponding actions
+ */
+public class Parser {
+    private final String command;
+    private final Ui ui;
+
+    /**
+     * Constructor for parser
+     * @param command user's inputted command
+     */
+    public Parser(String command) {
+        this.command = command;
+        this.ui = new Ui();
+    }
+
+    /**
+     * Parses the command and matches it with the corresponding action to be taken
+     * Executes the action on the task list given
+     * @param tasks ArrayList of current tasks stored on the chatbot
+     * @return boolean true if bye command entered and chatbot should be stopped, false otherwise
+     */
+    public String executeCommand(TaskList tasks, Storage storage) throws IOException {
+        String[] commands = command.strip().split(" ", 2);
+        try {
+            switch (commands[0]) {
```

Comment:
Good distinction for String[] commands as a multi-valued variable. 👍 
Very very small nitpick, you could name commands[0] as e.g. commandType to improve readability to show what commands[0] represents, but just personal preference. 😄